BistroQ.Presentation/Views/UserControls/Client/ProductListControl.xaml.cs

```diff
@@ -81,4 +83,27 @@ private void Button_PointerExited(object sender, PointerRoutedEventArgs e)
     {
         (sender as UIElement)?.ChangeCursor(CursorType.Arrow);
     }
+
+    private async void SingleProductControl_ProductClicked(object sender, ProductViewModel e)
+    {
+        var _dialogService = App.GetService<IDialogService>();
+
+        if (_dialogService != null)
```

Comment:
Sometimes the `DialogService` is null, I don't know why, may be it is not initialized correctly. 

If don't check null here, it may crash the program when view product detail.

Reply:
It seems like `AddScoped` behaves differently in non-web apps compared to web apps. Could the implicit behavior be causing the issue? Have you tried moving initialization inside the constructor?

According to [this](https://stackoverflow.com/questions/77494990/whats-the-difference-between-addscoped-and-addsingleton-in-net-core-console-ap)

Reply:
It does not help much. Now I can remove the dialog service. The bug that dialog not shown in the first login still happened.